packages/rebber-plugins/src/preprocessors/codeVisitor.js

```diff
@@ -1,41 +1,53 @@
 const visit = require('unist-util-visit')
+const appendix = require('../type/appendix')
 
 module.exports = (ctx, tree) => {
-  const title = ctx.codeAppendiceTitle || 'Appendices'
+  ctx.overrides.appendix = appendix
   return (node) => {
-    const inAppendix = []
-    let appendixIndex = 1
-
+    let appendix = tree.children[tree.children.length - 1]
+    if (!appendix || appendix.type !== 'appendix') {
+      appendix = {
+        type: 'appendix',
+        children: [],
+      }
+      tree.children.push(appendix)
+    }
+    let appendixIndex = appendix.children.length + 1
     visit(node, 'code', (innerNode, index, _parent) => {
-      inAppendix.push({
+
+      appendix.children.push({
         type: 'paragraph',
         children: [
           {
-            type: 'definition',
-            identifier: `appendix-${appendixIndex}`,
-            referenceType: 'full',
-            children: [{type: 'text', value: 'code'}],
+            type: 'heading',
+            children: [
+
+              {
+                type: 'definition',
+                identifier: `appendix-${appendixIndex}`,
+                url: `${ctx.codeAppendiceTitle || 'Appendix'} ${appendixIndex}`,
```

Comment:
Is `appendixIndex` really different from `index` here? I'm thinking these two should be equal according to the way you count.

Reply:
I don't understant what youare meaning here.

index is the index of current visited not in its parent. If the parent is a table cell, index will almost always be 0. Whereas we need the appendixIndex to be incremented each time we add a code into it